ITensorGaussianMPS/src/gmps.jl

```diff
@@ -292,16 +660,47 @@ function correlation_matrix_to_mps(
   Λ::AbstractMatrix;
   eigval_cutoff::Float64=1e-8,
   maxblocksize::Int=size(Λ, 1),
+  minblocksize::Int=1,
   kwargs...,
 )
-  @assert size(Λ, 1) == size(Λ, 2)
+  return correlation_matrix_to_mps(
+    s,
+    ConservingNf(Λ0);
+    eigval_cutoff=eigval_cutoff,
+    maxblocksize=maxblocksize,
+    minblocksize=minblocksize,
+    kwargs...,
+  )
+end
+
+function correlation_matrix_to_mps(
+  s::Vector{<:Index},
+  Λ0::T;
+  eigval_cutoff::Float64=1e-8,
+  maxblocksize::Int=size(Λ0.data, 1),
+  minblocksize::Int=1,
+  kwargs...,
+) where {T<:Union{Pairing,ConservingNf}}
+  if eltype(Λ0.data) <: AbstractFloat
+    MPS_Elt = Float64
+  else
+    MPS_Elt = ComplexF64
```

Comment:
Can't you just use `eltype(Λ0)` for the eltype of the MPS instead of hardcoding it to `Float64` or `ComplexF64`?